### PR TITLE
Add a createError variant without error code (NFC)

### PR DESCRIPTION
**Pull request by adrian-prantl**
For the significant amount of call sites that want to create an incontrovertible error, such a wrapper function creates a significant readability improvement and lowers the cost of entry to add error handling in more places.

### CODE REVIEW

**Comment by adrian-prantl** [comment]
The obvious downside is that it makes it easier to not even try to find a matching error code, but it looks like most incontrovertible errors are caught before the get reported as an exit status.

File: lldb/source/Host/common/Socket.cpp [hunk 1]
```diff
@@ -87,8 +87,8 @@ llvm::Error Socket::Initialize() {
   if (err == 0) {
     if (wsaData.wVersion < wVersion) {
       WSACleanup();
-      return llvm::make_error<llvm::StringError>(
-          "WSASock version is not expected.", llvm::inconvertibleErrorCode());
+      return llvm::createStringError("WSASock version is not expected.",
+                                     llvm::inconvertibleErrorCode());
```
Comment (JDevlieghere) [inline lldb/source/Host/common/Socket.cpp:91]: This doesn't need the `inconvertibleErrorCode` anymore, right?

**Review by JDevlieghere — COMMENTED** [review]
I'm in favor of this. I rarely have a meaningful error code and it makes the code needlessly verbose. 

I was going to ask if we wanted to make an overload of the existing `createStringError` but without the error code. But then I noticed that it uses the printf-style formatters (and not the llvm one). I think @bulbazord tried to add a `formatv` overload but that slowed down compilation of LLVM itself. Long story short, I think having this function take an `llvm::Twine` is fine.

File: lldb/source/Host/common/Socket.cpp (hunk 1 above)
Comment (adrian-prantl) [reply lldb/source/Host/common/Socket.cpp:91]: Yes, that's an oversight. I also didn't try to exhaustively convert everything, just give a sample of the kinds of expected improvements.

**Review by medismailben — APPROVED** [review]
This will be very useful. LGTM!

**Comment by lhames** [comment]
Late to the party here, but some context might be of interest:

`inconvertibleErrorCode` was meant to never be used, except with a documented explanation of why no error code could be supplied. The idea was to make conversion between `Error` and `std::error_code` illegal when no error code was supplied.

In practice (1) nobody has adhered to the documentation rule (me included), and (2) nobody seems to be converting errors back into std::error_codes in a dangerous way*. At this point I'd be fine getting rid of `inconvertibleErrorCode` entirely and replacing it with an "unknownErrorCode" that could be used as a default instead.

* There is some danger that people _have_ tried such invalid conversions and either caught the issue before submitting, or have never hit the dangerous path at runtime and don't realize it's there. If we switched to a safe "unknownErrorCode" we would lose some signal on the former (it'd be an error report rather than a crash), but would prevent the latter from happening in production.

**Comment by adrian-prantl** [comment]
Thanks for providing the context, Lang!
Inside of LLDB we use llvm::Expected often in completely closed environments where the expectation is always that all errors are handled internally and we never would exit the process with an error code, so this would be useful here.

**Pull request merged**